frontend/src/app/app.component.ts

```diff
@@ -37,12 +39,18 @@ export class AppComponent implements OnInit, AfterViewInit, AfterContentChecked
     constructor(private translateService: TranslateService,
                 public auth: AuthService,
                 public responsiveHelper: ResponsiveHelperService,
+                private router: Router,
                 /*private overlayContainer: OverlayContainer*/) {
     }
 
     ngOnInit() {
         this.recheckMobile();
         this.changeTheme(this.getThemeFromLocalStorage() || OCTheme.OCEAN);
+        this.router.events
+            .filter(event => event instanceof NavigationEnd)
+            .filter(() => isNotNull(this.sidenav))
```

Comment:
TBH I'm not sure if I like filter operations for something that does not interact with the Observable's elements, it seems like that would not be intended  (I'm sure it works though).
I would prefer an if statement in the last statement like this: 
```typescript
.do(() => 
    if (isNotNull(this.sidenav) && this.sidenav.opened) this.sidenav.close();
);
```
(I know I said I like it before but I changed my mind)